web/src/store.ts

```diff
@@ -8,37 +8,66 @@ export interface User {
 	isRestricted: boolean;
 }
 
-export interface Election {
-	roleName: string;
-	description: string;
+export interface Poll {
+	id: number;
 	isActive: boolean;
 	isConcluded: boolean;
-	id: number;
+	pollType: PollType;
+	election: Optional<Election>;
+	referendum: Optional<Referendum>;
 	candidates: Optional<Array<{ name: string; isMe: boolean }>>;
 }
 
+export interface PollType {
+	id: number;
+	name: string;
+}
+
+export interface Election {
```

Comment:
Consider extending Election from PollType using a discriminated union for additional type safety